gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/AccountProvider.java

```diff
@@ -63,34 +66,57 @@ public String path() {
+                activityAuditService,
+                accountManagementUserService,
+                factorManager,
+                applicationContext,
+                domain);
+        OAuth2AuthHandler oAuth2AuthHandler = OAuth2AuthHandler.create(oAuth2AuthProvider);
+        oAuth2AuthHandler.extractToken(true);
+        oAuth2AuthHandler.extractClient(true);
+        accountRouter.route().handler(oAuth2AuthHandler).handler(this::handleEnableCheck);
+
+        // user consent routes
+        accountRouter.get(AccountRoutes.INDEX.getRoute()).handler(accountHandler::getUserOrTimeout).handler(accountHandler::getAccount);
+        accountRouter.get(AccountRoutes.STATIC_ASSETS.getRoute()).handler(accountHandler::getUserOrTimeout).handler(accountHandler::getAsset);
+        accountRouter.get(AccountRoutes.PROFILE.getRoute()).handler(accountHandler::getUserOrTimeout).handler(accountHandler::getProfile);
+        accountRouter.put(AccountRoutes.PROFILE.getRoute()).handler(BodyHandler.create()).handler(accountHandler::getUserOrTimeout).handler(accountHandler::updateProfile);
+        accountRouter.get(AccountRoutes.ACTIVITIES.getRoute()).handler(accountHandler::getUserOrTimeout).handler(accountHandler::getActivity);
+        accountRouter.get(AccountRoutes.CHANGE_PASSWORD.getRoute()).handler(accountHandler::redirectForgotPassword);
+
+        accountRouter.get(AccountRoutes.FACTORS.getRoute()).handler(accountHandler::getUserOrTimeout).handler(accountHandler::getUserFactors);
+        accountRouter.get(AccountRoutes.FACTORS_CATALOG.getRoute()).handler(accountHandler::getUserOrTimeout).handler(accountHandler::getSecurityDomainFactors);
+        accountRouter.get(AccountRoutes.FACTORS_BY_ID.getRoute()).handler(accountHandler::getUserOrTimeout).handler(accountHandler::getUserEnrolledFactorById);
+        accountRouter.delete(AccountRoutes.FACTORS_BY_ID.getRoute()).handler(accountHandler::getUserOrTimeout).handler(accountHandler::removeFactorFromUser);
+        accountRouter.post(AccountRoutes.FACTORS.getRoute()).handler(BodyHandler.create()).handler(accountHandler::getUserOrTimeout).handler(accountHandler::enrollFactor);
+        accountRouter.post(AccountRoutes.FACTORS_VERIFY.getRoute()).handler(BodyHandler.create()).handler(accountHandler::getUserOrTimeout).handler(accountHandler::handleFactorVerify);
+        accountRouter.post(AccountRoutes.FACTORS_OTP_QR.getRoute()).handler(BodyHandler.create()).handler(accountHandler::getUserOrTimeout).handler(accountHandler::handleGetOtpQr);
+
+        // error handler
+        accountRouter.route().failureHandler(new ErrorHandler());
+        router.mountSubRouter(path(), accountRouter);
     }
 
     private boolean isSelfServiceAccountEnabled() {
         return domain.getSelfServiceAccountManagementSettings() != null && domain.getSelfServiceAccountManagementSettings().isEnabled();
     }
+
+    private void handleEnableCheck(RoutingContext req) {
+        if (isSelfServiceAccountEnabled()) req.fail(404);
```

Comment:
If enabled, the request should got to next handler else return a 404 but returning only the status will trigger a NullPointerException into `io.gravitee.am.gateway.handler.common.vertx.web.handler.ErrorHandler`